### PR TITLE
Bump version number to 0.13.0rc0

### PR DESCRIPTION
**Pull request by josh146**
**Context:** Prepares for the 0.13.0 release candidate 0 release.

**Description of the Change:**

* Bumps version number to 0.13.0rc0

* Updates changelog with recent PRs, and corrects Python example

* Pins `sphinx-gallery` to version 0.5, as 0.6 was recently released, breaking the documentation CSS.

**Benefits:** n/a

**Possible Drawbacks:** n/a

**Related GitHub Issues:** n/a

### CODE REVIEW

File: .github/CHANGELOG.md [hunk 1]
```diff
@@ -1,22 +1,31 @@
...
   ```python
-  from strawberryfields import RemoteEngine
+  import strawberryfields as sf
   from strawberryfields import ops
   from strawberryfields.utils import random_interferometer
 
   # replace AUTHENTICATION_TOKEN with your Xanadu cloud access token
-  eng = RemoteEngine("chip2", token="AUTHENTICATION_TOKEN")
+  con = sf.api.Connection(token="AUTH_TOKEN")
+  eng = sf.RemoteEngine("X8", connection=con)
```
Comment (josh146) [inline .github/CHANGELOG.md:28]: With the new API client, a connection object must be created if you wish to create the connection programmatically (as opposed to using `store_account` or `sf --configure`).

File: .github/CHANGELOG.md [hunk 2]
```diff
@@ -29,13 +38,14 @@
 
       ops.Interferometer(U) | q[:4]
       ops.Interferometer(U) | q[4:]
+      ops.MeasureFock() | q
 
-  result = eng.run(prog)
+  result = eng.run(prog, shots=1000)
```
Comment (josh146) [inline .github/CHANGELOG.md:43]: Shots is now a required argument when submitting to the remote engine.

File: .github/CHANGELOG.md [hunk 3]
```diff
@@ -1,22 +1,31 @@
-# Release 0.13.0-dev (development release)
+# Release 0.13.0rc0 (development release)
```
Comment (trbromley) [inline .github/CHANGELOG.md:1]:
```suggestion
# Release 0.13.0rc0 (release candidate)
```

File: .github/CHANGELOG.md [hunk 4]
```diff
@@ -1,22 +1,31 @@
...
 
   ```python
-  from strawberryfields import RemoteEngine
+  import strawberryfields as sf
   from strawberryfields import ops
   from strawberryfields.utils import random_interferometer
 
   # replace AUTHENTICATION_TOKEN with your Xanadu cloud access token
-  eng = RemoteEngine("chip2", token="AUTHENTICATION_TOKEN")
+  con = sf.api.Connection(token="AUTH_TOKEN")
```
Comment (antalszava) [inline .github/CHANGELOG.md:27]: Getting an `AttributeError` here:
```python
AttributeError                            Traceback (most recent call last)
<ipython-input-3-bf2d6bbfbb6e> in <module>
----> 1 sf.api.Connection()

AttributeError: module 'strawberryfields' has no attribute 'api'
```

Comment (antalszava) [inline .github/CHANGELOG.md:27]: (Edit: removed my previous comment here as I was using the wrong SF :sweat_smile: )

**Pull request merged**